### DISCUSSION TITLE
ESP32-S3 use both ECM and webusb reduce endpoints

### DISCUSSION THREAD

**Discussion by Jacques-Zhao** [hathach/tinyusb#1362] [Q&A] [answered]
Hi,
I am new to USB, I am using ESP32-S3 to develop my USB project. I have tested ECM and Webusb demo separately, but when I integrated them together, I found that ECM could not work. 
Through analysis, I found that ESP32-S3 only supports 6 additional endpoints (1-6) to work.
While ECM and Webusb work at the same time requires 8 endpoints, I found that Webusb itself needs 5 endpoints(2 vendor endpoints, 1 CDC notify and 2 CDC data endpoint), if I just want to transfer simple data, can I reuse or delete some endpoints?
Looking forward to your reply.

**Comment by Jacques-Zhao** [comment, marked as answer]
Sorry, When using Webusb function, the CDC is not required, and I removed the endpoints,now the test is fine.

**Comment by hathach** [comment]
one note is that you seems to misread the endpoints limit of esp32s2, it is as follows: 
- 5 endpoints IN including control ep0
- 6 endpoints OUT including control ep0
